craft_application/application.py

```diff
@@ -612,7 +614,13 @@ def _render_secrets(self, yaml_data: dict[str, Any]) -> None:
 
         self._secrets = secret_values
 
-    def _extra_yaml_transform(self, yaml_data: dict[str, Any]) -> dict[str, Any]:
+    def _extra_yaml_transform(
+        self,
+        yaml_data: dict[str, Any],
+        *,
+        build_on: str,  # noqa: ARG002 (Unused method argument)
+        build_for: str | None,  # noqa: ARG002 (Unused method argument)
+    ) -> dict[str, Any]:
```

Comment:
Is here missing logic or it is a NOOP? I see you override the `_extra_yaml_transform` in the test.

Reply:
It's a noop here, yeap. It's meant to be specialized by the applications